SwiftCompilerSources/Sources/Optimizer/InstructionSimplification/SimplifyMisc.swift

```diff
@@ -30,3 +30,21 @@ extension TypeValueInst: OnoneSimplifiable, SILCombineSimplifiable {
     context.erase(instruction: self)
   }
 }
+
+extension EndCOWMutationAddrInst: OnoneSimplifiable, SILCombineSimplifiable {
```

Comment:
Can you add this to a new file `SimplifyEndCOWMutationAddr.swift`? It's not good to put unrelated simplifications in a single file (I don't like `SimplifyMisc.swift` and we should rename it to `SimplifyTypeValue.swift`).
Also, can you add a short comment what kind of simplification is done?